### PR TITLE
FirebaseInstallations FID generation: don't cut UUID

### PR DESCRIPTION
**Pull request by maksymmalyhin**
- copy last 4 bits of UUID to the FID prefix (they were cut in the previous implementation).

### CODE REVIEW

**Comment by maksymmalyhin** [comment]
cc @andirayo, @ankitaj224

File: FirebaseInstallations/Source/Library/FIRInstallationsItem.m [hunk 1]
```diff
@@ -72,22 +72,22 @@ + (NSString *)generateFID {
 
   NSData *uuidData = [NSData dataWithBytes:uuidBytes length:16];
```
Comment (paulb777) [inline FirebaseInstallations/Source/Library/FIRInstallationsItem.m]: Make a define for the 16 - UUID_LENGTH perhaps?

File: FirebaseInstallations/Source/Library/FIRInstallationsItem.m [hunk 2]
```diff
@@ -72,22 +72,22 @@ + (NSString *)generateFID {
...
 
-  uint8_t prefix = 0b01110000;
-  NSMutableData *fidData = [NSMutableData dataWithBytes:&prefix length:1];
+  uint8_t uuidLast4Bits = uuidBytes[15] & 0b00001111;
```
Comment (paulb777) [inline FirebaseInstallations/Source/Library/FIRInstallationsItem.m]: UUID_LENGTH-1

Comment (chliangGoogle) [inline FirebaseInstallations/Source/Library/FIRInstallationsItem.m]: acronym should be capitalized. same as below

Comment (maksymmalyhin) [reply FirebaseInstallations/Source/Library/FIRInstallationsItem.m]: I defined a local var `NSUInteger UUIDLength = sizeof(uuid_t);`, so the UUIDBytes[UUIDLength - 1] would not crash even if actual `UUID` length changes.

**Pull request merged**